spanner/transaction_test.go

```diff
@@ -268,7 +268,6 @@ func TestReadWriteTransaction_ErrorReturned(t *testing.T) {
 	requests := drainRequestsFromServer(server.TestSpanner)
```

Comment:
We should add test cases for known corner cases with InlineBeginTransaction, so at least for the following cases:
1. Verify that everything works as expected if a read/write transaction executes two queries in parallel as the first statement. Only one should include a `Begin`, and the other should block until the first has returned a transaction.
2. What happens if the first statement in a transaction returns an error?
3. What happens if the first statement in a transaction is a BatchDML, and the first N statements succeed, but statement N+1 fails?
4. What happens if the first statement in a transaction is a Read (not Query)?
5. What happens if the first statement in a transaction is a query or read operation that initially succeeds, but then later fails with an `Unavailable` error while consuming the stream of `PartialResultSet`s? The request will be retried, but it should not include a `BeginTransaction` option, as a transaction was already returned by the initial request.